### ISSUE TITLE
Hadoop incompatible when using ADL UFS

### ISSUE THREAD

**Issue by LuQQiu** [Alluxio/alluxio#13377]
**Alluxio Version:**
Master 2.6.0-SNAPSHOT-ee131efc279e141a151c00f3c201fb934efc7b46

**Describe the bug**
Data is already distributed loaded into Alluxio.
ImageNet training job is listing and reading data through Alluxio Fuse mount point.
```
    -Dalluxio.worker.fuse.enabled=true -Dalluxio.worker.fuse.mount.alluxio.path=/ -Dalluxio.worker.fuse.mount.point=/mnt/alluxio-fuse 
    -Dalluxio.worker.fuse.mount.options=allow_other,kernel_cache,max_read=131072,attr_timeout=7200,entry_timeout=7200 
```

The training job can successfully pass the list directory phrase but will be stuck and eventually timeout in the reading data phrase. The stuck behavior happens sometimes at the beginning of the reading job (happened once) or after reading for about 30 minutes (happened twice). 

After the training job reading data timeout, running `cat` command in one of the workers will hang for a long time and eventually succeed.  The other workers don't have this behavior and are functioning normally. The worker has the hanging `cat` behavior has the following error message in their worker.log:
```
2021-05-05 08:02:25,463 ERROR BlockReadHandler - Failed to run DataReader.
com.google.common.util.concurrent.ExecutionError: java.lang.ExceptionInInitializerError
Caused by: java.lang.ExceptionInInitializerError
	at org.apache.hadoop.fs.azure.AzureNativeFileSystemStore$1.eventOccurred(AzureNativeFileSystemStore.java:2013)
Caused by: java.lang.NullPointerException
	at org.apache.hadoop.util.VersionInfo.<init>(VersionInfo.java:46)
	at org.apache.hadoop.util.VersionInfo.<clinit>(VersionInfo.java:99)
	... 43 more
2021-05-07 20:00:46,499 ERROR BlockReadHandler - Failed to run DataReader.
com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.util.VersionInfo
  at alluxio.worker.block.UnderFileSystemBlockReader.updateUnderFileSystemInputStream(UnderFileSystemBlockReader.java:314)
  at alluxio.worker.block.UnderFileSystemBlockReader.init(UnderFileSystemBlockReader.java:138)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.util.VersionInfo
  at org.apache.hadoop.fs.azure.AzureNativeFileSystemStore$1.eventOccurred(AzureNativeFileSystemStore.java:2013)
  at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:914)
```
Full useful worker log for the hanging `cat` worker [error.log](https://github.com/Alluxio/alluxio/files/6446721/error.log)

For this hanging `cat` worker:
```
    "Worker.BytesReadRemote.10.240.0.14": {
      "count": 0
    },
    "Worker.Fuse.BytesRead.10.240.0.14": {
      "count": 40817501
    },
    "Worker.Fuse.BytesToRead.10.240.0.14": {
      "count": 60370944
    },
```
There is no `BytesReadDirect`, and `BytesReadRemote` is 0, which means that nothing got to read from this worker. 
All other workers find BytesReadDirect and BytesReadRemote
```
    "Worker.BytesReadDirect.10.240.0.10": {
      "count": 274375900
    },
    "Worker.BytesReadRemote.10.240.0.10": {
      "count": 10439234567
    },
```
No deadlock found from the hanging `cat` worker jstack [jstack.log](https://github.com/Alluxio/alluxio/files/6462185/jstack.log)

**Comment by LuQQiu** [comment]
Looks like the default hadoop version and the ADL hadoop version is not compatible.

**Comment by LuQQiu** [comment]
Fixed by #13395
